src/js/components/Layer/LayerContainer.js

```diff
@@ -17,137 +17,120 @@ const HiddenAnchor = styled.a`
-
-    let content = (
-      <StyledContainer
+  }, [position]);
+
+  const theme = themeContext || propsTheme;
+
+  let content = (
+    <StyledContainer
+      id={id}
+      {...rest}
+      position={position}
+      plain={plain}
+      responsive={responsive}
+      ref={containerRef}
+    >
+      <HiddenAnchor ref={anchorRef} tabIndex="-1" aria-hidden="true" />
+      {children}
+    </StyledContainer>
+  );
+
+  if (modal) {
+    content = (
+      <StyledLayer
         id={id}
-        {...rest}
-        position={position}
         plain={plain}
+        position={position}
         responsive={responsive}
-        ref={this.containerRef}
+        tabIndex="-1"
+        ref={layerRef}
       >
-        {/* eslint-disable max-len */}
-        {/* eslint-disable jsx-a11y/anchor-is-valid, jsx-a11y/anchor-has-content */}
-        <HiddenAnchor ref={this.anchorRef} tabIndex="-1" aria-hidden="true" />
-        {/* eslint-enable jsx-a11y/anchor-is-valid, jsx-a11y/anchor-has-content */}
-        {/* eslint-enable max-len */}
-        {children}
-      </StyledContainer>
-    );
-
-    if (modal) {
-      content = (
-        <StyledLayer
-          id={id}
+        <StyledOverlay
           plain={plain}
-          position={position}
+          onMouseDown={onClickOutside}
           responsive={responsive}
-          tabIndex="-1"
-          ref={this.layerRef}
-        >
-          <StyledOverlay
-            plain={plain}
-            onMouseDown={onClickOutside}
-            responsive={responsive}
-          />
-          {content}
-        </StyledLayer>
-      );
-      /* eslint-enable jsx-a11y/anchor-is-valid, jsx-a11y/anchor-has-content */
-    }
-
-    if (onEsc) {
-      content = <Keyboard onEsc={onEsc}>{content}</Keyboard>;
-    }
+        />
+        {content}
+      </StyledLayer>
+    );
+  }
 
-    if (theme.layer.background) {
-      const dark = backgroundIsDark(theme.layer.background, theme);
-      if (dark !== theme.dark) {
-        content = (
-          <ThemeContext.Provider value={{ ...theme, dark }}>
-            {content}
-          </ThemeContext.Provider>
-        );
-      }
-    }
+  if (onEsc) {
+    content = <Keyboard onEsc={onEsc}>{content}</Keyboard>;
+  }
 
-    if (modal) {
+  if (theme.layer.background) {
+    const dark = backgroundIsDark(theme.layer.background, theme);
+    if (dark !== theme.dark) {
       content = (
-        <FocusedContainer hidden={position === 'hidden'} restrictScroll>
+        <ThemeContext.Provider value={{ ...theme, dark }}>
           {content}
-        </FocusedContainer>
+        </ThemeContext.Provider>
       );
     }
-    return content;
   }
-}
+
+  if (modal) {
+    content = (
+      <FocusedContainer hidden={position === 'hidden'} restrictScroll>
+        {content}
+      </FocusedContainer>
+    );
+  }
+  return content;
+};
+
+LayerContainer.defaultProps = {
+  full: false,
+  margin: 'none',
+  modal: true,
+  position: 'center',
+};
```

Comment:
Same comment here to move the assignment to be part of the declaration.

Reply:
Made the change.